cmis/models/cmis_backend.py

```diff
@@ -0,0 +1,121 @@
+    browsing_ok = fields.Boolean(
+        'Allow browsing this backend')
+    storing_ok = fields.Boolean(
+        'Allow storing in this backend')
+
+    @api.multi
+    def _get_base_adapter(self):
+        """
+        Get an adapter to test the backend connection
+        """
+        self.ensure_one()
+        session = ConnectorSession.from_env(self.env)
+        environment = ConnectorEnvironment(self, session, None)
+        return CmisAdapter(environment)
+
+    @api.multi
+    def check_auth(self):
+        """ Check the authentication with DMS """
+        self.ensure_one()
+        adapter = self._get_base_adapter()
+        return adapter._auth(self)
+
+    @api.multi
+    def check_directory_of_write(self):
+        """Check access right to write from the path"""
+        datas_fname = 'testdoc'
+        for this in self:
+            # login with the cmis account
+            repo = this.check_auth()
+            folder_path_write = this.initial_directory_write
+            # Testing the path
+            rs = repo.query("SELECT cmis:path FROM  cmis:folder")
+            bool_path_write = self.check_existing_path(rs, folder_path_write)
+            # Check if we can create a doc from OE to EDM
+            # Document properties
+            if bool_path_write:
+                sub = repo.getObjectByPath(folder_path_write)
+                try:
+                    sub.createDocumentFromString(
+                        datas_fname,
+                        contentString='hello, world',
+                        contentType='text/plain')
+                except cmislib.exceptions.UpdateConflictException:
+                    raise CMISError(
+                        _("The test file already exists in the DMS. "
+                          "Please remove it and try again."))
+                except cmislib.exceptions.RuntimeException:
+                    raise CMISError(
+                        ("Please check your access right."))
+            self.get_error_for_path(bool_path_write, folder_path_write)
+
+    @api.multi
+    def check_directory_of_read(self):
+        """Check access right to read from the path"""
+        for this in self:
+            repo = this.check_auth()
+            folder_path_read = this.initial_directory_read
+            # Testing the path
+            rs = repo.query("SELECT cmis:path FROM  cmis:folder ")
+            bool_path_read = self.check_existing_path(rs, folder_path_read)
+            self.get_error_for_path(bool_path_read, folder_path_read)
+
+    def check_existing_path(self, rs, folder_path):
+        """Function to check if the path is correct"""
+        for one_rs in rs:
+            # Print name of files
+            cmis_path = one_rs.getProperties()['cmis:path']
+            if folder_path == cmis_path or folder_path in cmis_path:
+                return True
+        return False
+
+    def get_error_for_path(self, is_valid, path):
```

Comment:
I would seperate this method in 2.

when you are calling the method, you already know what kind of exception you are expecting.